### ISSUE TITLE
BatchNorm test time code

### ISSUE THREAD

**Issue by Hazarapet** [fchollet/keras#6127]
Guys I've seen the #81 issue and as I understand you remember mean and variance during training time (when train=True), and at the test time you use previously remembered mean and variance (on the last step), But when I looked inside the code I saw something weird.
![rsz_keras](https://cloud.githubusercontent.com/assets/4758675/24618275/afa774e0-18a7-11e7-9309-db510faa54c0.png)

When **train=False**, it returns just **normed**, which is calculated for the inputs' batch, and only after that if (inside the else block) it calls K.batch_normalization(...) which uses T.nnet.bn.batch_normalization_test(...).Could you explain me please where you put remembered mean and variance over the last step's set?

**Comment by allanzelener** [comment]
Agreed that this seems to disagree with the expected behavior for BatchNormalization. The Keras 2.0 API release notes say: " BatchNorm now only supports mode 0 (use batch metrics for feature-wise normalization during training, and use moving metrics for feature-wise normalization during testing)." but the moving metrics don't seem to be used. 

I think the behavior should be changed to match the release notes' description since that matches the original paper's implementation and that the documentation for BN should be expanded to be more descriptive of these details on the layer's expected behavior. Not sure if I have time to make the PR myself but this seems like a high priority misspecification.

**Comment by Hazarapet** [comment]
Is everything clear in Keras 2.0? Does it work the way it should be?

**Comment by allanzelener** [comment]
Ah, Francois refactored the code yesterday and removed the possibility of the `normalize_batch_in_training ` from being returned when `training in {0, False}`. I'm not sure if that was actually happening since I haven't yet looked at how Keras models use the `training` flag but everything else makes sense now.

Keras 2.0 has already been merged to master and master now has a fix for this issue so it should be closed.

**Comment by Hazarapet** [comment]
The master of this repository is Keras 2.0 already? I thought it will be another branch.How long does the master branch contain Keras 2.0?

**Comment by allanzelener** [comment]
Since March 14. See release notes here: https://github.com/fchollet/keras/wiki/Keras-2.0-release-notes

**Comment by Hazarapet** [comment]
ok.Thanks
